api/internal/git/cloner.go

```diff
@@ -44,6 +45,19 @@ func ClonerUsingGitExec(repoSpec *RepoSpec) error {
 			repoSpec.CloneSpec(), repoSpec.Dir.String())
 	}
 
+	cmd = exec.Command(
+		gitProgram,
+		"fetch",
+		"--depth=1",
+		"origin",
+		repoSpec.Ref)
+	cmd.Dir = repoSpec.Dir.String()
+	out, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Printf("Error fetching ref: %s", out)
+		return errors.Wrapf(err, "trouble fetching %s", repoSpec.Ref)
+	}
+
 	cmd = exec.Command(
 		gitProgram,
 		"checkout",
```

Comment:
Checkout `FETCH_HEAD` here.

Reply:
Thank you for the catching this. I've pushed a fix for it.